components/brave_education/resources/brave_education.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { EducationPageProxy } from './education_page_proxy'
+import { readCommandMessage } from './message_data_reader'
+
+function pageReady() {
+  return new Promise<void>((resolve) => {
```

Comment:
this should use connectedCallback/disconnectedCallback like upstream. Having some kind of simplified version of upstream is definitely not what we want here. It should be essentially identical.

Reply:
I was not aware that copying upstream's Lit-based `whats_new` implementation was a requirement. Since the cost-benefit ratio of redoing the front-end is not clear to me, and it's purely a dev concern. I suggest that we take an iterative approach address this in a future PR. I'm happy to file an issue to capture this.

Reply:
Follow-up issue created here: https://github.com/brave/brave-browser/issues/42562